packages/block-editor/src/components/block-parent-selector/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockIcon from '../block-icon';
+
+/**
+ * Block parent selector component, displaying the hierarchy of the
+ * current block selection as a single icon to "go up" a level.
+ *
+ * @return {WPElement} Parent Block Selector.
+ */
+export default function BlockParentSelector() {
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const { parents } = useSelect( ( select ) => {
+		const { getSelectedBlockClientId, getBlockParents } = select(
+			'core/block-editor'
+		);
+		const selectedBlockClientId = getSelectedBlockClientId();
+		return {
+			parents: getBlockParents( selectedBlockClientId ),
+		};
+	}, [] );
+
+	const firstParentClientId = parents[ parents.length - 1 ];
+
+	const { parentBlockType } = useSelect( ( select ) => {
+		const { getBlockName } = select( 'core/block-editor' );
+		const parentBlockName = getBlockName( firstParentClientId );
+		return {
+			parentBlockType: getBlockType( parentBlockName ),
+		};
+	}, [] );
```

Comment:
As @youknowriad suggested, we can combine the `useSelect` calls.

```suggestion
	const { parents, parentBlockType } = useSelect( ( select ) => {
		const {
			getBlockName,
			getBlockParents,
			getSelectedBlockClientId
		} = select( 'core/block-editor' );
		const selectedBlockClientId = getSelectedBlockClientId();
		const _parents = getBlockParents( selectedBlockClientId )
		const firstParentClientId = _parents[ _parents.length - 1 ];
		const parentBlockName = getBlockName( firstParentClientId );

		return {
			parents: _parents,
			parentBlockType: getBlockType( parentBlockName ),
		};
	}, [] );
```

Reply:
I tried this one, but it breaks the editor. There's a `firstParentClientId not defined" error in the console.

Reply:
Whoops! I forgot that `firstParentClientId` was being used outside of the select callback.

This should work:
```suggestion
	const {
		parents,
		parentBlockType,
		firstParentClientId
	} = useSelect( ( select ) => {
		const {
			getBlockName,
			getBlockParents,
			getSelectedBlockClientId
		} = select( 'core/block-editor' );
		const selectedBlockClientId = getSelectedBlockClientId();
		const _parents = getBlockParents( selectedBlockClientId )
		const _firstParentClientId = _parents[ _parents.length - 1 ];
		const parentBlockName = getBlockName( _firstParentClientId );

		return {
			parents: _parents,
			parentBlockType: getBlockType( parentBlockName ),
			firstParentClientId: _firstParentClientId
		};
	}, [] );
```